src/include/duckdb/planner/tableref/bound_cteref.hpp

```diff
@@ -29,5 +30,7 @@ class BoundCTERef : public BoundTableRef {
 	idx_t bind_index;
 	//! The index of the cte
 	idx_t cte_index;
+	//! Is this a reference to a materialized CTE?
+	bool materialized_cte;
```

Comment:
Can we use the `CTEMaterialize` enum here?

Reply:
Yes, `CTEMaterialize` is used now.